db/collab.go

```diff
@@ -148,12 +195,7 @@ func (d *DB) JoinCollab(c echo.Context) error {
 			var msg Message
 
 			if err := websocket.JSON.Receive(ws, &msg); err != nil {
-				errorMsg := Message{
-					Author: uid,
-					Type:   "ERROR",
-					Body:   err.Error(),
-				}
-				if broadcastErr := session.Broadcast(errorMsg); broadcastErr != nil {
+				if err2 := session.BroadcastError(uid, err.Error()); err2 != nil {
```

Comment:
You can just reinitialize and declare `err` here. These vars are scope-local.

```suggestion
				if err := session.BroadcastError(uid, err.Error()); err2 != nil {
```